lib/worker.js

```
@@ -46,6 +46,46 @@ function Worker(options) {
   this.repopulate();
 }
 
+Worker.prototype.updateWorker = function(newOptions, callback){
+  if(newOptions){
+    if(newOptions.url){
+      this._url = newOptions.url;
+      this._queue.concurrency = newOptions.parallel || 2;
+      if (newOptions.paused === true) {
+        this.pause();
+      } else {
+        this.resume();
+      }
+      this.updateTx = newOptions.updateTx === true;
+      this.forwardMetadata = newOptions.forwardMetadata === true;
+
+      Winston.info('Worker updated');
+      Winston.debug({
+        url:this._url,
+        parallel: this._queue.concurrency,
+        updateTx: this.updateTx,
+        forwardMetadata:this.forwardMetadata
+      });
+    } else {
+      callback('Failed to update worker: url is required')
```

For these I think it's better to return an Error object - `callback(new Error('Failed to update worker: url is required'));`.